src/main/java/org/interledger/cryptoconditions/Fulfillment.java

```diff
@@ -1,13 +1,36 @@
 package org.interledger.cryptoconditions;
 
-public interface Fulfillment {
+/**
+ * An implementation of a crypto-conditions Fulfillment.
+ *
+ * @see "https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/"
+ */
+public interface Fulfillment<C extends Condition> {
 
-  ConditionType getType();
+  /**
+   * Accessor for the type of this fulfillment.
+   *
+   * @return A {@link CryptoConditionType} for this fulfillment.
+   */
+  CryptoConditionType getType();
 
-  byte[] getEncoded();
```

Comment:
As with Condition, this corresponds to https://docs.oracle.com/javase/7/docs/api/java/security/Key.html#getEncoded()

The idea is that a condition could be seen as a public key and the fulfillment as a private key and signature